app/custom_domain_utils.py

```diff
@@ -89,12 +97,14 @@ def sanitize_domain(domain: str) -> str:
     return new_domain
 
 
-def can_domain_be_used(user: User, domain: str) -> Optional[CannotUseDomainReason]:
+def can_domain_be_used(
```

Comment:
Although code reuse is good I'd split this into three functions:
 - one that has all the logic without taking into account custom domain or blocked domain
 - one that calls the previous and checks also for duplicate custom domains 
 - one that calls the first and checks that this user doesn't have any duplicate blocked domain